src/nvenc/nvenc_base.cpp

```diff
@@ -293,6 +298,30 @@ namespace nvenc {
         fill_vui(format_config.hevcVUIParameters);
         break;
       }
+
+      case 2: {
+        // AV1
+        auto &format_config = enc_config.encodeCodecConfig.av1Config;
+        format_config.repeatSeqHdr = 1;
+        format_config.idrPeriod = NVENC_INFINITE_GOPLENGTH;
+        format_config.chromaFormatIDC = 1;  // YUV444 not supported by NVENC yet
+        if (buffer_is_10bit()) {
+          format_config.inputPixelBitDepthMinus8 = 2;
+          format_config.pixelBitDepthMinus8 = 2;
+        }
+        format_config.colorPrimaries = colorspace.primaries;
+        format_config.transferCharacteristics = colorspace.tranfer_function;
+        format_config.matrixCoefficients = colorspace.matrix;
+        format_config.colorRange = colorspace.full_range;
+        set_ref_frames(format_config.maxNumRefFramesInDPB, format_config.numBwdRefs, 8);

```

Comment:
`numFwdRefs` here. It's not worded very well, but that's the correct one. Number of reference frames used in forward prediction is probably what they meant (and it mentions "golden" frames, while the other one mentions L1).

Max ref frames is 7. "The AV1 codec allows a maximum of 8 frames in its
decoded frame buffer. For a coding frame, it can choose any 7
frames from the decoded frame buffer as its reference frames".

Reply:
And I'm not sure AV1 allows to use more than 4 of those 7 for forward prediction, need to test this with RFI.

Reply:
Re-reading that paper again, 8 is probably the correct number. 7 is the limit one frame can use from the pool 8: 4 for forward prediction, and 3 for backwards.

Reply:
Or it's 7, with the intention of currently decoded frame to replace the 8-th. And god knows what NvENC means by their "maxNumRefFramesInDPB". 7 should be the safe number, I would go with 7 for now (while adding the comment to find a way to confirm or deny 8).

Reply:
I think your initial quote is correct - https://aomedia.org/av1/specification/symbols/

AFAIK, the DPB allows 8 reference frames (`NUM_REF_FRAMES`) but only 7 (`REFS_PER_FRAME`) can be referenced by any single frame (4 forward and 3 backward) plus an intra reference against the frame itself for 8 (`TOTAL_REFS_PER_FRAME`).

I've tested for several hours streaming with packet loss and didn't see any artifacts with RFI on AV1, so we either have the correct value of 8 or Nvidia is clamping it to something lower. I tried passing something obviously bogus like 900 and NVENC still initialized and worked, so I don't think we can distinguish these two. In either case though, it's doing the right thing at runtime.

Reply:
I think we can test it by manually calling RFI to see when it produces an IDR frame. But it's not a blocker and can be done in the future.